subsys/bluetooth/host/hci_core.c

```diff
@@ -3810,11 +3810,15 @@ uint16_t bt_get_appearance(void)
 int bt_set_appearance(uint16_t appearance)
 {
 	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
-		int err = settings_save_one("bt/appearance", &appearance, sizeof(appearance));
-
-		if (err) {
-			BT_ERR("Unable to save setting 'bt/appearance' (err %d).", err);
-			return err;
+		if(bt_dev.appearance != appearance)
+		{
```

Comment:
you could even move the rest of the function inside that if.

Reply:
@jori-nordic , you are right!

Fixed also code guideline issues in last pull request